actors/builtin/multisig/multisig_actor.go

```diff
@@ -99,15 +97,15 @@ func (a *MultiSigActor) Propose(rt vmr.Runtime, params *ProposeParams) *ProposeR
 
 	// Note: this ID may not be stable across chain re-orgs.
 	// https://github.com/filecoin-project/specs-actors/issues/7
-	return &ProposeReturn{txnID}
+	out := cbg.CborInt(txnID)
+	return &out
 }
 
 type TxnIDParams struct {
 	ID TxnID
 }
 
 func (a *MultiSigActor) Approve(rt vmr.Runtime, params *TxnIDParams) *vmr.EmptyReturn {
-	rt.ValidateImmediateCallerType(builtin.CallerTypesSignable...)
```

Comment:
This is a significant item coming from the spec, the source already of much debate, and still here by fiat from jbenet. We can't just remove it, but your runtime shim can no-op it for now if you want.

Reply:
It's not doing anything productive. Its checking that the call is coming from some specific types of actors, when theres no good reason for that. So what if some new type of actor wants to interact with a multisig? The multisig explicitly doesnt care

Reply:
I 100% agree with you but already lost this battle. Please restore it here for now so we can land these changes and then discuss elsewhere.

Reply:
*sigh* fine